File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length - 1;
        while ( i < j) {
            int curr = numbers[i] + numbers[j];
            if(curr == target) return new int[] { i+1, j+1};
            else if (curr > target) j--;
            else i++;
        }
        return new int[2];
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = {}
        res= []
        for i in range(len(nums)):
            if target - nums[i] in s:
                if i < s.get(target - nums[i]):
                    res = [ i,s.get(target - nums[i])]
                else:
                    res = [s.get(target - nums[i]),i]
            s[nums[i]] = i

        return res